src/vsg/app/Viewer.cpp

```diff
@@ -464,7 +464,7 @@ void Viewer::setupThreading()
     size_t numEarlyTransferTasks = 0;
     for (auto& task : recordAndSubmitTasks)
     {
-        if (task->commandGraphs.size() >= 1)
+        if (!task->commandGraphs.empty())
```

Comment:
Using empty() instead of comparing the size() with 0 is a good practice. The price of getting the size depends on the container type. For vectors it is O(1), but for lists is O(n). empty() is always O(1) and you don't need to check container type or take care if it will be changed in the future.

Check, for example, usage of size() / empty() in src/vsg/utils/Builder.cpp and compare complexity of used vsg::Array::size() / vsg::Array::empty()

Reply:
Good taste in programming is more important than general rules.  Readability trumps general rules.  The containers where size() is being used are O(1) and will never be any different, so it's silly to use that as a reason.  

So every instance we need to look at the code and figure out what is most efficient and most readable.  General rules are not absolute.  To try and apply rules as if they are absolutes leads to bad programming.

Read the code, does it make sense.  Does changing the code make it code make it better?  Not does it follow to rule x or rule y, does the code change make the code better?  If you it's not a clear improvement then it's really just a style preference not an actual code quality issue,

If I am pushing back then it's a style preference not a quality issue.  All code changes we make code break the build or break the runtime so just changing things on a whim/prefernece is really not something we should be doing.

Reply:
I personally prefer the “is (not empty)” statement. It’s just a bit more clear to read. Just my 2c.

Reply:
I understand your position concerning such PRs.

Possibly I have too critical opinion concerning static code analysis tools after years of working in projects with zero tolerance to LINT warnings and strict MISRA rules. But I still consider such things like static code analysis tools or strict coding style guidelines useful to make code cleaner and more consistent. It also require additional discipline from contributors.

OK, I'll extract parts related to bug fixes and create another PR this night.